### ISSUE TITLE
Every time VS Code starts the terminal reports "installing c# dependencies"

### ISSUE THREAD

**Issue by gmccullo** [OmniSharp/omnisharp-vscode#5015]
## Issue Description ##

Every time VS Code starts the terminal reports "installing c# dependencies"

## Steps to Reproduce ##

Install C# VS Code.

## Expected Behavior ##

No terminal output.

## Actual Behavior ##

Terminal opens and reports:

```
Installing C# dependencies...
Platform: win32, x86_64

Finished
```

### OmniSharp log ###

Starting OmniSharp server at 1/21/2022, 11:35:47 AM
    Target: c:\Users\gary\Dropbox\Private\Stories\Cygnus\analyzers\Analyzers\Analyzers.sln

OmniSharp server started.
    Path: c:\Users\gary\.vscode\extensions\ms-dotnettools.csharp-1.24.0\.omnisharp\1.38.0\OmniSharp.exe
    PID: 6220

[info]: OmniSharp.Stdio.Host
        Starting OmniSharp on Windows 6.2.9200.0 (x64)
[info]: OmniSharp.Services.DotNetCliService
        Checking the 'DOTNET_ROOT' environment variable to find a .NET SDK
[info]: OmniSharp.Services.DotNetCliService
        Using the 'dotnet' on the PATH.
[info]: OmniSharp.Services.DotNetCliService
        DotNetPath set to dotnet
[info]: OmniSharp.MSBuild.Discovery.MSBuildLocator
        Located 2 MSBuild instance(s)
            1: Visual Studio Build Tools 2019 16.11.32106.194 16.11.2 - "C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\MSBuild\Current\Bin"
            2: StandAlone 17.0.0 - "c:\Users\gary\.vscode\extensions\ms-dotnettools.csharp-1.24.0\.omnisharp\1.38.0\.msbuild\Current\Bin"
[warn]: OmniSharp.CompositionHostBuilder
        It looks like you have Visual Studio 2019 installed without .NET Core SDK support which is required by OmniSharp.
 Try updating Visual Studio 2019 installation with .NET Core SDK to enable better MSBuild support.
[info]: OmniSharp.MSBuild.Discovery.MSBuildLocator
        MSBUILD_EXE_PATH environment variable set to 'c:\Users\gary\.vscode\extensions\ms-dotnettools.csharp-1.24.0\.omnisharp\1.38.0\.msbuild\Current\Bin\MSBuild.exe'
[info]: OmniSharp.MSBuild.Discovery.MSBuildLocator
        Registered MSBuild instance: StandAlone 17.0.0 - "c:\Users\gary\.vscode\extensions\ms-dotnettools.csharp-1.24.0\.omnisharp\1.38.0\.msbuild\Current\Bin"
            CscToolExe = csc.exe
            CscToolPath = c:\Users\gary\.vscode\extensions\ms-dotnettools.csharp-1.24.0\.omnisharp\1.38.0\.msbuild\Current\Bin\Roslyn
            MSBuildExtensionsPath = c:\Users\gary\.vscode\extensions\ms-dotnettools.csharp-1.24.0\.omnisharp\1.38.0\.msbuild
            MSBuildToolsPath = c:\Users\gary\.vscode\extensions\ms-dotnettools.csharp-1.24.0\.omnisharp\1.38.0\.msbuild\Current\Bin
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.CSharpFormattingWorkspaceOptionsProvider, Order: 0
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.Completion.CompletionOptionsProvider, Order: 0
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.RenameWorkspaceOptionsProvider, Order: 100
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.ImplementTypeWorkspaceOptionsProvider, Order: 110
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.BlockStructureWorkspaceOptionsProvider, Order: 140
[info]: OmniSharp.Cake.CakeProjectSystem
        Detecting Cake files in 'c:\Users\gary\Dropbox\Private\Stories\Cygnus\analyzers\Analyzers'.
[info]: OmniSharp.Cake.CakeProjectSystem
        Did not find any Cake files
[info]: OmniSharp.MSBuild.ProjectSystem
        Detecting projects in 'c:\Users\gary\Dropbox\Private\Stories\Cygnus\analyzers\Analyzers\Analyzers.sln'.
[info]: OmniSharp.MSBuild.ProjectManager
        Queue project update for 'c:\Users\gary\Dropbox\Private\Stories\Cygnus\analyzers\Analyzers\Analyzers\Analyzers.csproj'
[info]: OmniSharp.MSBuild.ProjectManager
        Queue project update for 'c:\Users\gary\Dropbox\Private\Stories\Cygnus\analyzers\Analyzers\Linkify\Linkify.csproj'
[info]: OmniSharp.MSBuild.ProjectManager
        Queue project update for 'c:\Users\gary\Dropbox\Private\Stories\Cygnus\analyzers\Analyzers\LinkifyTest\LinkifyTest.csproj'
[info]: OmniSharp.MSBuild.ProjectManager
        Queue project update for 'c:\Users\gary\Dropbox\Private\Stories\Cygnus\analyzers\Analyzers\analyzerslib\AnalyzersLib.csproj'
[info]: OmniSharp.MSBuild.ProjectManager
        Queue project update for 'c:\Users\gary\Dropbox\Private\Stories\Cygnus\analyzers\Analyzers\AnalyzersLibTest\AnalyzerTest.csproj'
[info]: OmniSharp.Script.ScriptProjectSystem
        Detecting CSX files in 'c:\Users\gary\Dropbox\Private\Stories\Cygnus\analyzers\Analyzers'.
[info]: OmniSharp.Script.ScriptProjectSystem
        Did not find any CSX files
[info]: OmniSharp.WorkspaceInitializer
        Configuration finished.
[info]: OmniSharp.Stdio.Host
        Omnisharp server running using Stdio at location 'c:\Users\gary\Dropbox\Private\Stories\Cygnus\analyzers\Analyzers' on host 6436.
[info]: OmniSharp.MSBuild.ProjectManager
        Loading project: c:\Users\gary\Dropbox\Private\Stories\Cygnus\analyzers\Analyzers\Analyzers\Analyzers.csproj
[info]: OmniSharp.MSBuild.ProjectManager
        Successfully loaded project file 'c:\Users\gary\Dropbox\Private\Stories\Cygnus\analyzers\Analyzers\Analyzers\Analyzers.csproj'.
[info]: OmniSharp.MSBuild.ProjectManager
        Adding project 'c:\Users\gary\Dropbox\Private\Stories\Cygnus\analyzers\Analyzers\Analyzers\Analyzers.csproj'
[info]: OmniSharp.MSBuild.ProjectManager
        Loading project: c:\Users\gary\Dropbox\Private\Stories\Cygnus\analyzers\Analyzers\Linkify\Linkify.csproj
[info]: OmniSharp.MSBuild.ProjectManager
        Successfully loaded project file 'c:\Users\gary\Dropbox\Private\Stories\Cygnus\analyzers\Analyzers\Linkify\Linkify.csproj'.
[info]: OmniSharp.MSBuild.ProjectManager
        Adding project 'c:\Users\gary\Dropbox\Private\Stories\Cygnus\analyzers\Analyzers\Linkify\Linkify.csproj'
[info]: OmniSharp.MSBuild.ProjectManager
        Loading project: c:\Users\gary\Dropbox\Private\Stories\Cygnus\analyzers\Analyzers\LinkifyTest\LinkifyTest.csproj
[info]: OmniSharp.MSBuild.ProjectManager
        Successfully loaded project file 'c:\Users\gary\Dropbox\Private\Stories\Cygnus\analyzers\Analyzers\LinkifyTest\LinkifyTest.csproj'.
[info]: OmniSharp.MSBuild.ProjectManager
        Adding project 'c:\Users\gary\Dropbox\Private\Stories\Cygnus\analyzers\Analyzers\LinkifyTest\LinkifyTest.csproj'
[info]: OmniSharp.MSBuild.ProjectManager
        Loading project: c:\Users\gary\Dropbox\Private\Stories\Cygnus\analyzers\Analyzers\analyzerslib\AnalyzersLib.csproj
[info]: OmniSharp.MSBuild.ProjectManager
        Successfully loaded project file 'c:\Users\gary\Dropbox\Private\Stories\Cygnus\analyzers\Analyzers\analyzerslib\AnalyzersLib.csproj'.
[info]: OmniSharp.MSBuild.ProjectManager
        Adding project 'c:\Users\gary\Dropbox\Private\Stories\Cygnus\analyzers\Analyzers\analyzerslib\AnalyzersLib.csproj'
[info]: OmniSharp.MSBuild.ProjectManager
        Loading project: c:\Users\gary\Dropbox\Private\Stories\Cygnus\analyzers\Analyzers\AnalyzersLibTest\AnalyzerTest.csproj
[info]: OmniSharp.MSBuild.ProjectManager
        Successfully loaded project file 'c:\Users\gary\Dropbox\Private\Stories\Cygnus\analyzers\Analyzers\AnalyzersLibTest\AnalyzerTest.csproj'.
[info]: OmniSharp.MSBuild.ProjectManager
        Adding project 'c:\Users\gary\Dropbox\Private\Stories\Cygnus\analyzers\Analyzers\AnalyzersLibTest\AnalyzerTest.csproj'
[info]: OmniSharp.MSBuild.ProjectManager
        Update project: Analyzers
[info]: OmniSharp.MSBuild.ProjectManager
        Update project: Linkify
[info]: OmniSharp.MSBuild.ProjectManager
        Update project: LinkifyTest
[info]: OmniSharp.MSBuild.ProjectManager
        Update project: AnalyzersLib
[info]: OmniSharp.MSBuild.ProjectManager
        Update project: AnalyzerTest

### C# log ###

Installing C# dependencies...
Platform: win32, x86_64

Finished

## Environment information ##

**VSCode version**: 1.63.2
**C# Extension**: 1.24.0

<details><summary>Dotnet Information</summary>
.NET SDK (reflecting any global.json):
 Version:   6.0.101
 Commit:    ef49f6213a

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.22000
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\6.0.101\

Host (useful for support):
  Version: 6.0.1
  Commit:  3a25a7f1cc

.NET SDKs installed:
  6.0.101 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 6.0.1 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 3.1.22 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.1 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.22 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.1 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

To install additional .NET runtimes or SDKs:
  https://aka.ms/dotnet-download
</details>
<details><summary>Visual Studio Code Extensions</summary>

|Extension|Author|Version|
|---|---|---|
|block-travel|sashaweiss|0.0.2|
|Bookmarks|alefragnani|13.2.2|
|csharp|ms-dotnettools|1.24.0|
|docomment|k--kato|0.1.30|
|duplicate-finder|tlevesque2|0.0.6|
|ecdc|mitchdenny|1.4.0|
|fix-json|oliversturm|0.1.2|
|fluent-icons|miguelsolorio|0.0.16|
|font-switcher|evan-buss|4.0.3|
|html-preview-vscode|tht13|0.2.5|
|HTMLHint|mkaufman|0.10.0|
|icons-carbon|antfu|0.2.4|
|insert-unicode|brunnerh|0.13.1|
|luahelper|yinfei|0.2.10|
|markdown-all-in-one|yzhang|3.4.0|
|markdown-links|tchayen|0.8.0|
|markdown-shortcuts|mdickin|0.12.0|
|markdown-toc|CharlesWan|1.5.6|
|material-icon-theme|PKief|4.11.0|
|material-product-icons|PKief|1.1.1|
|MDTools|seanmcbreen|1.0.1|
|overtype|adammaras|0.2.0|
|powershell|ms-vscode|2021.12.0|
|prettier-vscode|esbenp|9.1.0|
|render-crlf|medo64|1.5.22|
|rtf|alefragnani|2.4.0|
|test-adapter-converter|ms-vscode|0.1.5|
|todo-tree|Gruntfuggly|0.0.215|
|vsc-material-theme-icons|equinusocio|2.2.1|
|vscode-autohotkey|slevesque|0.2.2|
|vscode-dotnet-adapter|derivitec-ltd|1.4.3|
|vscode-fileutils|sleistner|3.5.0|
|vscode-icons|vscode-icons-team|11.8.0|
|vscode-markdownlint|DavidAnson|0.45.0|
|vscode-open-file-folder|auchenberg|0.0.4|
|vscode-sort-json|richie5um2|1.20.0|
|vscode-table-formatter|shuworks|1.2.1|
|vscode-test-explorer|hbenl|2.21.1|
|vscode-w3cvalidation|Umoxfo|2.9.1|
|winteriscoming|johnpapa|1.4.4|
|word-counter|plus5keen|0.0.1|
|xml|DotJoshJohnson|2.5.1|;

</details>

**Comment by Sov3rain** [comment]
Same behaviour here, same versions. Annoying but harmless.

**Comment by ubxtrol** [comment]
I have the same problem when I upgrade to version 1.24.0 (C# Extension).

**Comment by heshuimu** [comment]
There is a way to get rid of that message: 

1. Go into your user settings and set "useModernNet" to true. Restart VS Code. 
2. Open a folder with a C# project to activate the C# plugin, watch it installs the new OmniSharp. Once done, close the Output panel, and restart VS Code. 
3. The message should no longer appear and the Output panel should not show up automatically on the next launch once you close it. 

The reason seems to be how the new logic handles install. Now because of the "useModernNet" stuff, when the setting is set to false, the plugin would skip installing the new OmniSharp. But because it is a part of the dependency list, and the person did not include logic to remove it from the dependency list and still pass in the entire dependency list, so at every launch C# plugin would try to install it again, but then new logic would skip it, again. 
https://github.com/OmniSharp/omnisharp-vscode/blob/752a56a7a079beeeb5bc36860a4b1fa0b659fabb/src/main.ts#L231-L233
https://github.com/OmniSharp/omnisharp-vscode/blob/752a56a7a079beeeb5bc36860a4b1fa0b659fabb/src/InstallRuntimeDependencies.ts#L13-L17
https://github.com/OmniSharp/omnisharp-vscode/blob/752a56a7a079beeeb5bc36860a4b1fa0b659fabb/src/packageManager/downloadAndInstallPackages.ts#L25-L28

**Comment by realivanjx** [comment]
can i leave `useModernNet` to true @heshuimu ? would there be a problem if i dont set it back to false? thanks.

**Comment by heshuimu** [comment]
> can i leave `useModernNet` to true @heshuimu ? would there be a problem if i dont set it back to false? thanks.

You can if you are not doing Unity development. See the README file regarding the change.

**Comment by ameltzerMSFT** [comment]
This output window behavior is *incredibly* obnoxious. I see this every single time I open VSCode. It's good that `useModernNet` can be used to work around this but this should really be fixed in omnisharp so this doesn't keep happening.

**Comment by ameltzerMSFT** [comment]
Setting `useModernNet` to true doesn't seem to have really helped at all. The "Output" window still pops up every time I open  VSCode except now it's just empty.

**Comment by heshuimu** [comment]
> Setting `useModernNet` to true doesn't seem to have really helped at all. The "Output" window still pops up every time I open  VSCode except now it's just empty.

VS Code remembers what panel is opened, so you have to close it yourself. It has never popped up for me ever since.

**Comment by pyrareae** [comment]
I can confirm this is also happening to me, and it's grating on my nerves. Especially since I primarily don't even use C#.

**Comment by hermitcat** [comment]
Same here! I'm shaking and crying rn

**Comment by duyongoc** [comment]
same here! Annoying :(

**Comment by abluescarab** [comment]
Chiming in with the same issue and the same annoyance. I frequently debug programs in the terminal so I'd like it to actually stay on the tab I choose.

**Comment by filipw** [comment]
This was resolved in https://github.com/OmniSharp/omnisharp-vscode/pull/5041

**Comment by deanrih** [comment]
Thank you, it is harmless yet annoying.

Are there any words when this particular fix will be pushed out as a release and into VS Code?

**Comment by ovilaro** [comment]
It has been fixed since the latest version of the extension.

**Comment by filipw** [comment]
Released in 1.24.1
